community/bolt/src/main/java/org/neo4j/bolt/logging/BoltMessageLoggerImpl.java

```diff
@@ -191,7 +203,11 @@ private String getCorrelationId()
 
     private static String remoteAddress( Channel channel )
     {
-        return channel.remoteAddress().toString();
+        InetSocketAddress remoteAddress = (InetSocketAddress) channel.remoteAddress();
```

Comment:
Could this cast fail?

Reply:
It would never fail because we always create a SocketChannel. Having said that Never say Never. I have fixed it by handling that as an edge case.